### PR TITLE
`sage --package create`: When re-creating with a different source type, clean up

### PR DESCRIPTION
**Pull request by mkoeppe**
For example, when switching from a `pip` package to `wheel` package, remove the `requirements.txt` file.
When switching from `normal` to `wheel`, remove the `spkg-install.in` file.

One of these cases was handled by a commit on #36641, cherry-picked from there.
Generalizing here to avoid mistakes such as https://github.com/sagemath/sage/pull/37129#issuecomment-1943854621 in the future.

Example:
```
$ ./sage -package create imagesize --pypi
Downloading tarball from https://pypi.io/packages/py2.py3/i/imagesize/imagesize-1.4.1-py2.py3-none-any.whl to /Users/mkoeppe/s/sage/sage-rebasing/worktree-pristine/upstream/imagesize-1.4.1-py2.py3-none-any.whl
[......................................................................]
$ git status
On branch sage_package_create_remove_files
Your branch is ahead of 'upstream/develop' by 3 commits.
  (use "git push" to publish your local commits)

Changes not staged for commit:
  (use "git add/rm <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
	modified:   build/pkgs/imagesize/SPKG.rst
	modified:   build/pkgs/imagesize/checksums.ini
	modified:   build/pkgs/imagesize/install-requires.txt
	deleted:    build/pkgs/imagesize/spkg-install.in
	modified:   build/pkgs/imagesize/type
```

### :memo: Checklist

- [x] The title is concise, informative, and self-explanatory.
- [x] The description explains in detail what this PR is about.
- [ ] I have linked a relevant issue or discussion.
- [ ] I have created tests covering the changes.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: build/sage_bootstrap/creator.py [hunk 1]
```diff
@@ -80,6 +80,17 @@ def heading(title, char='-'):
...
                 f.write('{0}\n\n'.format(upstream_contact))
 
+    def _remove_files(self, files):
+        """
+        Remove ``files`` from the package directory if they exist
+        """
+        for file in files:
+            try:
+                # Remove this file, which would mark the package as a pip package.
+                os.remove(os.path.join(self.path, file))
```
Comment (kwankyu) [inline build/sage_bootstrap/creator.py:89]: delete this comment?

File: build/sage_bootstrap/creator.py [hunk 2]
```diff
@@ -80,6 +80,17 @@ def heading(title, char='-'):
             if upstream_contact:
                 f.write('{0}\n\n'.format(upstream_contact))
 
+    def _remove_files(self, files):
+        """
+        Remove ``files`` from the package directory if they exist
+        """
```
Comment (kwankyu) [inline build/sage_bootstrap/creator.py:86]: Period `.`

File: build/sage_bootstrap/creator.py [hunk 3]
```diff
@@ -101,13 +114,23 @@ def set_python_data_and_scripts(self, pypi_package_name=None, source='normal'):
...
             with open(os.path.join(self.path, 'install-requires.txt'), 'w+') as f:
                 f.write('{0}\n'.format(pypi_package_name))
+            # Remove this file, which would mark the package as a pip package.
+            self._remove_files(['requirements.txt'])
+            if pypi_package_name != 'pip':
+                # 'pip' should be the only wheel package that has a custom spkg-install.in script.
+                # Remove the script for all other wheel packages, to avoid errors when
+                # switching from normal to wheel packages.
+                self._remove_files(['spkg-build.in', 'spkg-install.in'])
         elif source == 'pip':
```
Comment (kwankyu) [inline build/sage_bootstrap/creator.py:128]: No `spkg-install` here?

File: src/doc/en/developer/packaging.rst [hunk 4]
```diff
@@ -92,6 +92,10 @@ the following source types:
 
    - its version number is defined by the required file ``package-version.txt``;
 
+   - No build and install scripts are needed
+     (with one exception: the package :ref:`spkg_pip` installs itself from
```
Comment (kwankyu) [inline src/doc/en/developer/packaging.rst:96]: `No` or `no`? Above, it is "its".

File: build/sage_bootstrap/creator.py [hunk 5]
```diff
@@ -101,13 +113,23 @@ def set_python_data_and_scripts(self, pypi_package_name=None, source='normal'):
...
+            if pypi_package_name != 'pip':
+                # 'pip' should be the only wheel package that has a custom spkg-install.in script.
+                # Remove the script for all other wheel packages, to avoid errors when
+                # switching from normal to wheel packages.
+                self._remove_files(['spkg-build.in', 'spkg-install.in', 'spkg-install'])
         elif source == 'pip':
             with open(os.path.join(self.path, 'requirements.txt'), 'w+') as f:
                 f.write('{0}\n'.format(pypi_package_name))
+            self._remove_files(['checksums.ini', 'spkg-build.in', 'spkg-install.in', 'spkg-install', 'install-requires.txt'])
         elif source == 'script':
```
Comment (kwankyu) [inline build/sage_bootstrap/creator.py:132]: No possibility of `spkg-build`?

Comment (mkoeppe) [reply build/sage_bootstrap/creator.py:132]: No, we don't have that.

Comment (kwankyu) [reply build/sage_bootstrap/creator.py:132]: OK.

**Review by kwankyu — APPROVED** [review]
LGTM.

**Comment by mkoeppe** [comment]
Thank you!

**Pull request merged**